patches/dmd/0004-fix-Issue-20441-Wrong-code-with-O-fPIC-and-pointer-s.patch

```diff
@@ -0,0 +1,58 @@
+From a4bc86ce3a0c4122fe27c84ae801c42f1c8edeef Mon Sep 17 00:00:00 2001
+From: Walter Bright <walter@walterbright.com>
+Date: Fri, 13 Dec 2019 00:54:42 -0800
+Subject: [PATCH] fix Issue 20441 - Wrong code with -O -fPIC and pointer
+ subtraction
+
+---
+ src/dmd/backend/cod2.d  |  2 ++
+ test/runnable/testpic.d | 16 ++++++++++++++++
+ 2 files changed, 18 insertions(+)
+
+diff --git a/src/dmd/backend/cod2.c b/src/dmd/backend/cod2.c
+index eecbba919..107127bbc 100755
+--- a/src/dmd/backend/cod2.c
++++ b/src/dmd/backend/cod2.c
+@@ -690,6 +690,8 @@ void cdorth(CodeBuilder& cdb,elem *e,regm_t *pretregs)
+         break;
+ 
+     case OPrelconst:
++        if (I64 && (config.flags3 & CFG3pic || config.exe == EX_WIN64))
++                goto L2;
```

Comment:
it falls through the same code than `Default`, right?

Reply:
Yes. The code is:
```
   case OPind:                                 /* if addressing mode   */
        if (!e2->Ecount)                        /* if not CSE           */
                goto L1;                        /* try OP reg,EA        */
        /* FALL-THROUGH */
    default:                                    /* operator node        */
    L2:
```
there is no `goto default` in C, so it needs to be `goto L2`.